llvm/utils/TableGen/SearchableTableEmitter.cpp

```diff
@@ -540,10 +569,12 @@ void SearchableTableEmitter::emitGenericTable(const GenericTable &Table,
 
   // Indexes are sorted "{ Thing, PrimaryIdx }" arrays, so that a binary
   // search can be performed by "Thing".
-  if (Table.PrimaryKey)
-    emitLookupFunction(Table, *Table.PrimaryKey, true, OS);
+  if (Table.PrimaryKey) {
+    auto &Index = Table.PrimaryKey;
```

Comment:
Why do we need this reference?

Reply:
This was needed before when an additional argument `ShouldReturnRange` was being passed to emitLookupFunction. But since, it was not needed anymore, we can remove this extra assignment statement as well. I have pushed a new patchset fixing the same